pkg/plugins/k8s/k8s_plugin.go

```diff
@@ -232,30 +254,53 @@ func (p *K8sPlugin) readManifestFiles() error {
 
 func (p *K8sPlugin) switchdevServiceStateUpdate() error {
 	// Check switchdev service
-	swdService, err := p.serviceManager.ReadService(p.switchdevService.Path)
+	swdService, err := p.serviceManager.ReadService(p.switchdevBeforeNMService.Path)
```

Comment:
we could avoid some duplicate logic here if we defined a method: `bool ShouldUpdateService(Service)`

and call it for both services

Reply:
Done, I used `isSwitchdevServiceNeedUpdate` as function name to be consistent with existing `isSystemServiceNeedUpdate`.